06.bowling_object/frame.rb

```diff
@@ -1,47 +1,57 @@
 # frozen_string_literal: true
 
 class Frame
-  attr_reader :shots, :score
+  attr_reader :shots
 
   def initialize(shots, frame_number)
     @shots = shots
     @frame_number = frame_number
   end
 
   def calculate_score(frames)
-    next_frame = next_frame(frames)
-    after_next_frame = after_next_frame(frames)
-
-    current_frame_score_sum = @shots.sum(&:point)
-    next_frame_score = next_frame.shots.map(&:point) if next_frame
-    after_next_frame_score = after_next_frame.shots.map(&:point) if after_next_frame
+    next_frame = frames[@frame_number + 1]
+    after_next_frame = frames[@frame_number + 2]
 
     if next_frame.nil?
-      current_frame_score_sum
-    elsif @shots[0].strike?
-      if next_frame.shots[0].strike? && after_next_frame
-        current_frame_score_sum + next_frame_score[0] + after_next_frame_score[0]
-      else
-        current_frame_score_sum + next_frame_score[0] + next_frame_score[1]
-      end
-    elsif spare?
-      current_frame_score_sum + next_frame_score[0]
+      current_frame_points_sum
     else
-      current_frame_score_sum
+      current_frame_points_sum + bonus_points(next_frame, after_next_frame)
     end
   end
 
-  def spare?
-    @shots.sum(&:point) == 10
+  private
+
+  def current_frame_points_sum
```

Comment:
current_frame = self なので、

```suggestion
  def points_sum
```

で良いと思います。